### PR TITLE
Add timeout while waiting for webserver to start

### PR DESCRIPTION
**Pull request by bjgordon**
In `citest.sh` I wanted to run htmlproofer before pa11y, as it is faster and so will find the things most likely to break sooner.

pa11y requires the webserver to be running before it starts. This is started in the background in parallel by circleci. I have moved the wait for the webserver later in the script as well so we're not waiting while we can do other work. We are using `set -x` (echo each line of the script), so this now means that every time we sleep a line is printed, which breaks [circleci timeouts](https://circleci.com/docs/configuration/#modifiers). Hence I've had to add a timeout.

I've done this with a simple loop counter.

### CODE REVIEW

File: bin/ciserve.sh [hunk 1]
```diff
@@ -15,4 +15,4 @@ if [ -z "$TMPDIR" ]; then
 fi
 
 # Start a local copy of the site to test against
-bundle exec jekyll serve --destination $TMPDIR/_pa11y_site
+bundle exec jekyll serve --destination $TMPDIR/_pa11y_site  >> _site/_serve_output.txt 2>&1
```
Comment (tessereth) [inline bin/ciserve.sh]: `>>` redirects stderr to the file. Then `2>&1` redirects stderr to stdout. Did you mean to redirect stdout to the file? If so, use a single `>`.

File: bin/citest.sh [hunk 2]
```diff
@@ -35,3 +26,18 @@ bundle exec htmlproofer _site  \
     --file-ignore /.*feed/index\.html/ \
     --empty-alt-ignore
 
+echo "Waiting for webserver to start..."
+COUNT=0
+until $(curl --output /dev/null --silent --head --fail http://localhost:4000); do
+    COUNT=$(($COUNT+1))
```
Comment (tessereth) [inline bin/citest.sh]: If you want to, you can use `$((COUNT++))`

File: bin/ciserve.sh (hunk 1 above)
Comment (bjgordon) [inline bin/ciserve.sh]: I will just remove this. I started to add it to debug why the test was failing.

File: bin/citest.sh (hunk 2 above)
Comment (bjgordon) [inline bin/citest.sh]: ✨

**Comment by bjgordon** [comment]
Thanks @tessereth I've implemented your comments

**Pull request merged**